python/ray/serve/_private/proxy_state.py

```diff
@@ -30,35 +32,174 @@
 logger = logging.getLogger(SERVE_LOGGER_NAME)
 
 
+class ActorWrapper(ABC):
+    def __init__(self, actor_handle: ActorHandle):
+        self._actor_handle = actor_handle
+        self._ready_obj_ref = None
+        self._health_check_obj_ref = None
+        self._is_drained_obj_ref = None
+        self._update_draining_obj_ref = None
+
+    @property
+    def actor_id(self) -> str:
+        return self._actor_handle._actor_id.hex()
+
+    @property
+    def actor_handle(self) -> ActorHandle:
+        return self._actor_handle
+
+    @property
+    def ready_obj_ref(self):
+        return self._ready_obj_ref
+
+    @property
+    def health_check_obj_ref(self):
+        return self._health_check_obj_ref
+
+    @property
+    def is_drained_obj_ref(self):
+        return self._is_drained_obj_ref
+
+    def clear_health_check_obj_ref(self):
+        self._health_check_obj_ref = None
+
+    def clear_drained_obj_ref(self):
+        self._is_drained_obj_ref = None
```

Comment:
"obj_ref" is entirely an implementation detail. it seems that these methods are also making assumptions about how the subclass will implement the health checks?

Reply:
Haha I was mostly just refactoring those objects/ methods into the wrapper and not thinking too much about the callers. Will drop all the obj_ref suffix and make them more tailed about the usecase.